core/algorithms/onpolicy_sync/engine.py

```diff
@@ -417,27 +449,23 @@ def collect_rollout_step(self, rollouts: RolloutStorage, visualizer=None) -> int
             rewards, dtype=torch.float, device=self.device,  # type:ignore
         )
 
-        # We want rewards to have dimensions [step, sampler, agent, reward]
+        # We want rewards to have dimensions [sampler, reward]
         if len(rewards.shape) == 1:
             # Rewards are of shape [sampler,]
-            rewards = rewards.view(1, -1, 1, 1)
-        elif len(rewards.shape) == 2:
-            # Rewards are of shape [sampler, agent]
-            rewards = rewards.unsqueeze(0).unsqueeze(-1)
+            rewards = rewards.unsqueeze(-1)
         else:
-            raise NotImplementedError
+            raise NotImplementedError()
 
         # If done then clean the history of observations.
         masks = torch.tensor(
-            [[0.0] if done else [1.0] for done in dones],
+            [0.0 if done else 1.0 for done in dones],
             dtype=torch.float32,
             device=self.device,  # type:ignore
-        )
+        ).view(
+            -1, 1
+        )  # [sampler, 1]
 
-        # Expand masks with new agents dimension
-        num_task_samplers = masks.shape[0]
-        num_agents = rewards.shape[2]
-        masks = masks.view(1, num_task_samplers, 1, 1).expand(-1, -1, num_agents, -1)
+        # self.probe(dones)
```

Comment:
Since it's mostly for debugging it seems fine to have it commented out. Maybe add a quick comment there explaining that `self.probe(dones)` is useful for debugging so someone (probably me) doesn't accidentally delete it thinking it's unnecessary.

Reply:
👍